modules/gitops-release-notes-1-17-0.adoc

```diff
@@ -0,0 +1,170 @@
+
+[id="gitops-1-17-0-fixed-regular-for-repeated-flags-in-extraCommandArgs_{context}"]
+=== Fixed regular expression handling in `sourceNamespaces`
+
+* Before this update, incorrect regular expression processing in the `sourceNamespaces` field caused improper labeling of namespaces. As a result, the `argocd.argoproj.io/managed-by-cluster-argocd` label was not applied, affecting namespace management. With this update, Argo CD correctly processes regular expressions in `sourceNamespaces`. The {gitops-title} Operator correctly applies labels based on regex patterns, enabling management of multiple namespaces using a single expression. link:https://issues.redhat.com/browse/GITOPS-6675[GITOPS-6675]
+
+[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]
+=== Fixed UI error when editing HTTP-based repository credentials
+
+* Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]
+
+[id="deprecated-features-1-17-0_{context}"]
+== Deprecated and removed features
+
+[id="deprecation-of-some-fields-in-argo-cd_{context}"]
+=== Deprecation of Keycloak support in OpenShift GitOps
+
+* In {gitops-title} 1.17, the support for Keycloak-based authentication is deprecated. When Keycloak is configured for single sign-on (SSO), deprecation warnings appear in the logs. Dex is the default and recommended authentication provider. It integrates with the OpenShift OAuth server and supports authentication using OpenShift users and groups. Users are required to migrate to Dex-based authentication for continued support. To continue using Keycloak, you must deploy and manage their own link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/operator_guide/installation-[Red Hat Build of Keycloak (RHBK)] instance and manually configure the integration in the Argo CD custom resource. link:https://issues.redhat.com/browse/GITOPS-7070[GITOPS-7070]
+
+[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
+=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields
+
+* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]
+
+[id="breaking-changes-1-17-0_{context}"]
```

Comment:
@anandf For breakng changes I think we should mention the actual updates instead of mentioning doc updates in upstream

Reply:
@varshab1210 Do you mean the intialRepositories breaking change here ?

Reply:
Let's incorporate latest suggestions made by @varshab1210

Reply:
Yes, these are Actual breaking changes and not documentation changes. @Dhruv-Soni11 Can you correct it.

Reply:
Fixed.